pkg/cmd/util/clientcmd/factory.go

```diff
@@ -192,9 +192,17 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 				return nil, errors.New("provided options object is not a BuildLogOptions")
 			}
 			if bopts.Version != nil {
-				// should --version work with builds at all?
 				return nil, errors.New("cannot specify a version and a build")
 			}
+			if bopts.Previous {
```

Comment:
One more case for server-side would be to insulate the clients from a hypothetical future changes to the meaning of "previous".